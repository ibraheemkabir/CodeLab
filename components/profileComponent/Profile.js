import React, { Component } from 'react';
import { Image, View,StyleSheet,Text } from 'react-native';
import { Card, CardItem,Button,Icon } from 'native-base';
import { TouchableWithoutFeedback, TouchableHighlight } from 'react-native-gesture-handler';

export default class CardImageExample extends Component {
	render() {
		const { navigate,data } = this.props
		return (
				<View>
					<Card>
						<CardItem cardBody style={{flexDirection: 'row',width: null}}>
						<Image source={{ uri: data.node.avatarUrl }} style={{ height: 380, width: null, flex: 1 }} />
						</CardItem>
					<View style={{
						flex: 1, alignItems: 'center', justifyContent: 'center', position: 'absolute', width: 410, backgroundColor: 'rgba(0, 0, 0, 0.85)'}}>
							<CardItem 
							cardBody 
							style={{
								flex: 1,
								width: 500,
								flexDirection: 'row', 
								width: 150, 
								alignItems: 'center', 
								justifyContent: 'center',
								height: 380,
								backgroundColor: 'transparent'
							}}
							>
	
							</CardItem>
						</View>
						<View style={{
							flex: 1, alignItems: 'center', justifyContent: 'center'
						}}>
							<CardItem cardBody
								style={{
									width: 150,
									alignItems: 'center',
									justifyContent: 'center',
							}}>
							<Image source={{ uri: data.node.avatarUrl}} style={{ height: 150, width: 150, flex: 1 , marginTop: -80}} />
							</CardItem>
						</View>
					<View>
				</View>
			</Card>
				<View style={{marginTop: 60}}>
					<View style={{ alignItems: 'flex-end', fontFamily: 'roboto', fontWeight: '600', color: '#47525E', marginTop: -50, paddingRight: 25 }}>
						<TouchableHighlight onPress={() => navigate('Web', { login: data.node.login })}>
							<Image source={require('../../assets/images/share-symbol.png')}  />	
						</TouchableHighlight>
					</View>
					<View style={{ alignItems:'center',textAlign:'center'}}>
							<Text style={{ fontSize: 22, fontFamily: 'roboto', fontWeight: '600', color: '#47525E'}}>
								{data.node.name}
							</Text>
							<Text style={{ fontSize: 20, fontFamily: 'roboto', fontWeight: '400', color: '#47525E',marginTop: 15 }}>
								{data.node.bio}
							</Text>
					</View>
				</View>
				<View style={{ marginTop: 30, alignItems: 'center', textAlign: 'center' }}>
					<View style={{ alignItems: 'center', textAlign: 'center', flexDirection: 'row',margin: 10}}>
						<Text style={{ fontSize: 20, fontFamily: 'roboto', letterSpacing: 1, fontWeight: '400', color: '#8492A6' }}>
							<Text style={{ fontSize: 20, fontWeight: '700', color: '#47525E' }}>{data.node.following.totalCount}</Text> following
						</Text>
						<Text style={{ fontSize: 15, fontFamily: 'roboto', fontWeight: '400', color: '#47525E',justifyContent: 'space-around',padding: 10 }}>
						|
						</Text>
						<Text style={{ fontSize: 20, fontFamily: 'roboto', letterSpacing: 1, fontWeight: '400', color: '#8492A6'}}>
							<Text style={{ fontSize: 20, fontWeight: '700', color: '#47525E' }}>{data.node.starredRepositories.totalCount}</Text>  Stars
						</Text>
					</View>
				</View>
			<View style={{
				flex: 1,
				flexDirection: 'column',
				justifyContent: 'space-between'
			}}>
					<View
						cardBody
						style={{
							flex: 1,
							flexDirection: 'row',
							justifyContent: 'space-between',
							padding: 30,
						}}>
						<Button style={styles.button}>
							<Text style={styles.btnText}>
								{data.node.repositories.totalCount}
									</Text>
							<Text style={styles.minibtnText}>
								Repositories
									</Text>
						</Button>
						<Button style={styles.button}>
							<Text style={styles.btnText}>
								{data.node.followers.totalCount}
									</Text>
							<Text style={styles.minibtnText}>
								Followers
									</Text>
						</Button>

					</View>
			</View>
		</View>
		);
	}
}

const styles = StyleSheet.create({
		button: {
		justifyContent: 'center', 
		alignContent: 'center', 
		backgroundColor: '#E18D96', 
		height: 200,
		width: 150, 
		backgroundColor: 'transparent',
		flexDirection: 'column',
		},
		btnText: {
			fontFamily: 'roboto',
			flex: 1, 
			textAlign: 'center', 
			fontWeight: '700', 
			color: '#47525E',
			fontSize: 85,
			letterSpacing: 1,
		},
		minibtnText: {
		fontFamily: 'roboto',
		flex: 1,
		textAlign: 'center',
		fontWeight: '700',
		color: '#47525E',
		fontSize: 17,
		letterSpacing: 1
	}
})