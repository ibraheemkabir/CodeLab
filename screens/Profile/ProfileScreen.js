import React,{Component} from 'react';
import {View} from 'react-native';
import {Icon} from 'native-base';
import Profile from '../../components/profileComponent/Profile'
import Modal from '../../components/modalComponent/modal'
export default class SettingsScreen extends Component {
  state = {
    modalVisible: false,
  };

  static navigationOptions = ({navigation}) => {
    const some = navigation.getParam('setModalVisibles')
    return{
      headerStyle: {
          backgroundColor: 'transparent',
          color: 'white',
          borderStyle: 'dotted',
          padding: 200
        },
        headerBackTitle:'hello',
        headerRight: (
          <Icon type="FontAwesome" name="share-alt" style={{ fontSize: 25, paddingRight: 20, color: 'white' }} onPress={() => some(true)} />
        ),
        headerTitleStyle: {
          color: 'white',
        },
        headerBackStyle: {
          color: 'white'
        },
        headerTintColor: 'white'
      }
  };

  setModalVisible = (visible) => {
    this.setState({ modalVisible: visible });
  }

  componentWillMount() {
    this.props.navigation.setParams({ setModalVisibles: this.setModalVisible });
  }

  render(){
    return (
      <View style={{ marginTop: -60 }}>
        <Modal setModalVisible={this.setModalVisible} modalVisible={this.state.modalVisible}/>
        <Profile />
      </View>
    );
  }
}

