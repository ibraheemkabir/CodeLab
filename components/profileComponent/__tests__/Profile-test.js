import React from 'react';
import renderer from 'react-test-renderer';

import Profile from '../Profile';

it(`renders correctly`, () => {
  const profile = renderer.create(<Profile/>);

  expect(profile).toMatchSnapshot();
});
