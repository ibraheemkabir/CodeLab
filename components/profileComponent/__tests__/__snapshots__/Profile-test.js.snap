// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      cardBody={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "flexDirection": "row",
            "paddingBottom": -5,
            "paddingLeft": -5,
            "paddingRight": -5,
            "paddingTop": -5,
          },
          Object {
            "flexDirection": "row",
            "width": null,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 380,
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, 0.85)",
          "flex": 1,
          "justifyContent": "center",
          "position": "absolute",
          "width": 410,
        }
      }
    >
      <View
        cardBody={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": -5,
              "paddingLeft": -5,
              "paddingRight": -5,
              "paddingTop": -5,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "flexDirection": "row",
              "height": 380,
              "justifyContent": "center",
              "width": 150,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        cardBody={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": -5,
              "paddingLeft": -5,
              "paddingRight": -5,
              "paddingTop": -5,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 150,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Object {
              "flex": 1,
              "height": 150,
              "marginTop": -80,
              "width": 150,
            }
          }
        />
      </View>
    </View>
    <View />
  </View>
  <View
    style={
      Object {
        "marginTop": 60,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#47525E",
            "fontFamily": "roboto",
            "fontSize": 22,
            "fontWeight": "600",
            "marginTop": -50,
            "paddingRight": 25,
          }
        }
      >
        <Text />
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#47525E",
            "fontFamily": "roboto",
            "fontSize": 22,
            "fontWeight": "600",
          }
        }
      >
        Brendan Moore
      </Text>
      <Text
        style={
          Object {
            "color": "#47525E",
            "fontFamily": "roboto",
            "fontSize": 20,
            "fontWeight": "400",
            "marginTop": 15,
          }
        }
      >
        CTO at Marvel Prototyping
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 30,
        "textAlign": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8492A6",
            "fontFamily": "roboto",
            "fontSize": 20,
            "fontWeight": "400",
            "letterSpacing": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#47525E",
              "fontSize": 20,
              "fontWeight": "700",
            }
          }
        >
          2390
        </Text>
          followers
      </Text>
      <Text
        style={
          Object {
            "color": "#47525E",
            "fontFamily": "roboto",
            "fontSize": 15,
            "fontWeight": "400",
            "justifyContent": "space-around",
            "padding": 10,
          }
        }
      >
        |
      </Text>
      <Text
        style={
          Object {
            "color": "#8492A6",
            "fontFamily": "roboto",
            "fontSize": 20,
            "fontWeight": "400",
            "letterSpacing": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#47525E",
              "fontSize": 20,
              "fontWeight": "700",
            }
          }
        >
          2390
        </Text>
          Stars
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      cardBody={true}
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 30,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "column",
            "height": 200,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#47525E",
              "flex": 1,
              "fontFamily": "Droid Sans Mono",
              "fontSize": 85,
              "fontWeight": "700",
              "letterSpacing": 1,
              "textAlign": "center",
            }
          }
        >
          123
        </Text>
        <Text
          style={
            Object {
              "color": "#47525E",
              "flex": 1,
              "fontFamily": "Droid Sans Mono",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 1,
              "textAlign": "center",
            }
          }
        >
          Repositories
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "column",
            "height": 200,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#47525E",
              "flex": 1,
              "fontFamily": "Droid Sans Mono",
              "fontSize": 85,
              "fontWeight": "700",
              "letterSpacing": 1,
              "textAlign": "center",
            }
          }
        >
          58
        </Text>
        <Text
          style={
            Object {
              "color": "#47525E",
              "flex": 1,
              "fontFamily": "Droid Sans Mono",
              "fontSize": 17,
              "fontWeight": "700",
              "letterSpacing": 1,
              "textAlign": "center",
            }
          }
        >
          Followers
        </Text>
      </View>
    </View>
  </View>
</View>
`;
